packages/astro/test/cli.test.js

```diff
@@ -32,28 +38,52 @@ describe('astro cli', () => {
 		expect(messages[0]).to.contain('started in');
 	});
 
-	const hostnames = [undefined, '0.0.0.0', '127.0.0.1'];
+	['dev', 'preview'].forEach((cmd) => {
+		const networkLogFlags = [['--host'], ['--host', '0.0.0.0'], ['--hostname', '0.0.0.0']];
+		networkLogFlags.forEach(([flag, flagValue]) => {
+			it(`astro ${cmd} ${flag} ${flagValue ?? ''} - network log`, async () => {
+				const { local, network } = await cliServerLogSetupWithFixture(flagValue ? [flag, flagValue] : [flag], cmd);
+
+				expect(local).to.not.be.undefined;
+				expect(network).to.not.be.undefined;
 
-	hostnames.forEach((hostname) => {
-		const hostnameArgs = hostname ? ['--hostname', hostname] : [];
-		it(`astro dev ${hostnameArgs.join(' ') || '(no --hostname)'}`, async () => {
-			const projectRootURL = new URL('./fixtures/astro-basic/', import.meta.url);
-			const proc = cli('dev', '--project-root', fileURLToPath(projectRootURL), ...hostnameArgs);
+				const localURL = new URL(local);
+				const networkURL = new URL(network);
+
+				expect(localURL.hostname).to.be.equal(flagValue ?? 'localhost', `Expected local URL to be on localhost`);
+				// Note: our tests run in parallel so this could be 3000+!
+				expect(Number.parseInt(localURL.port)).to.be.greaterThanOrEqual(3000, `Expected Port to be >= 3000`);
+				expect(networkURL.port).to.be.equal(localURL.port, `Expected local and network ports to be equal`);
+				expect(isIPv4(networkURL.hostname)).to.be.equal(true, `Expected network URL to respect --host flag`);
+			});
+		});
 
-			const { messages } = await parseCliDevStart(proc);
 
-			const local = messages[1].replace(/Local\s*/g, '');
-			const network = messages[2].replace(/Network\s*/g, '');
+		const hostToExposeFlags = [['', ''], ['--hostname', 'localhost']];
+		hostToExposeFlags.forEach(([flag, flagValue]) => {
+			it(`astro ${cmd} ${flag} ${flagValue} - host to expose`, async () => {
+				const { local, network } = await cliServerLogSetupWithFixture([flag, flagValue], cmd);
+
+				expect(local).to.not.be.undefined;
+				expect(network).to.not.be.undefined;
+				const localURL = new URL(local);
+
+				expect(localURL.hostname).to.be.equal('localhost', `Expected local URL to be on localhost`);
+				expect(() => new URL(networkURL)).to.throw();
+			});
+		});
+
+		const noNetworkLogFlags = [['--host', 'localhost'], ['--host', '127.0.0.1'], ['--hostname', '127.0.0.1']];
+		noNetworkLogFlags.forEach(([flag, flagValue]) => {
+			it(`astro ${cmd} ${flag} ${flagValue} - no network log`, async () => {
+				const { local, network } = await cliServerLogSetupWithFixture([flag, flagValue], cmd);
 
-			expect(local).to.not.be.undefined;
-			expect(network).to.not.be.undefined;
-			const localURL = new URL(local);
-			const networkURL = new URL(network);
+				expect(local).to.not.be.undefined;
+				expect(network).to.be.undefined;
```

Comment:
This is failing because the `parseCliDevStart` util is pretty naive.

Reply:
Fixed in [cce64e8](https://github.com/withastro/astro/pull/2760/commits/cce64e8102c485cbd6bb58d762c4b0a86857b359)